assert/_diff.ts

```diff
@@ -95,6 +95,14 @@ export function diff<T>(A: T[], B: T[]): Array<DiffResult<T>> {
     { length: size },
     () => ({ y: -1, id: -1 }),
   );
+  function getPoint(index: number) {
+    const point = fp[index];
+    if (!point) {
+      throw Error(`Missing FarthestPoint at index ${index}`);
+    }
+    return point;
+  }
```

Comment:
Using this new function to make lookups to fp safe adds a 10% performance overhead. I'm not sure if this is acceptable for this module so I wanted to raise it:

BENCH
```typescript
Deno.bench("_diff", () => {
  diff(["abc", "c"], ["abc", "bcd", "c"]);
});
```
BEFORE
```shell
file:///Users/simon.holloway/Code/deno_std/assert/_diff.ts
benchmark      time (avg)        iter/s             (min … max)       p75       p99      p995
--------------------------------------------------------------- -----------------------------
_diff         198.63 ns/iter   5,034,517.5 (178.23 ns … 679.03 ns) 193.23 ns 662.74 ns 667.21 ns
```
AFTER
```shell
file:///Users/simon.holloway/Code/deno_std/assert/_diff.ts
benchmark      time (avg)        iter/s             (min … max)       p75       p99      p995
--------------------------------------------------------------- -----------------------------
_diff         219.05 ns/iter   4,565,144.2 (188.39 ns … 730.58 ns) 217.06 ns 680.21 ns  691.8 ns
```

Reply:
Maybe ignore this, this benchmark seems to fluctuate by ~10% between calls. Not sure if there is actually any impact here.

Reply:
Oh, looks like I missed some tests, turns out throwing an error when fp[index] is undefined causes the tests to fail. I need to reconsider how to handle this case.